### PR TITLE
tikv-ctl: add bottommost level compaction

### PR DESCRIPTION
**Pull request by huachaohuang**
## What have you changed? (mandatory)

Add a bottommost level compaction option for the compact command to make it more flexible.

## What are the type of the changes? (mandatory)

- New feature (non-breaking change which adds functionality)

## How has this PR been tested? (mandatory)

TODO

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

TODO

## Does this PR affect tidb-ansible update? (mandatory)

No.

## Refer to a related PR or issue link (optional)

https://github.com/pingcap/kvproto/pull/284
https://github.com/pingcap/rust-rocksdb/pull/212

### CODE REVIEW

**Comment by siddontang** [comment]
any benefit for this feature? Can we compact more quickly?

**Comment by huachaohuang** [comment]
Well, we don't use compaction filter, so the bottommost level is skipped by default, which means this will not make us compact more quickly.
But we can force to compact the bottommost level for better compression, although it takes more time of course.

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -600,13 +611,30 @@ impl DebugExecutor for DebugClient {
...
+    fn do_compaction(
+        &self,
+        db: DBType,
+        cf: &str,
+        from: &[u8],
+        to: &[u8],
+        threads: u32,
+        bottommost: DBBottommostLevelCompaction,
+    ) {
+        let bottommost = match bottommost {
```
Comment (Hoverbear) [inline src/bin/tikv-ctl.rs]: Please extract this to a `From` or `Into`.

File: src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -1804,3 +1871,11 @@ fn compact_whole_cluster(
         h.join().unwrap();
     }
 }
+
+fn parse_bottommost_level_compaction(v: Option<&str>) -> DBBottommostLevelCompaction {
```
Comment (Hoverbear) [inline src/bin/tikv-ctl.rs]: Please extract this to a `From` or `Into`.

File: src/server/service/debug.rs [hunk 3]
```diff
@@ -228,6 +229,15 @@ impl<T: RaftStoreRouter + 'static + Send> debugpb_grpc::Debug for Service<T> {
 
     fn compact(&self, ctx: RpcContext, req: CompactRequest, sink: UnarySink<CompactResponse>) {
         let debugger = self.debugger.clone();
+
+        let bottommost = match req.get_bottommost_level_compaction() {
```
Comment (Hoverbear) [inline src/server/service/debug.rs]: Please extract this to a `From` or `Into`.

**Pull request merged**